src-java/FixedHiveDriver.java

```diff
@@ -0,0 +1,47 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.metabase.hive.jdbc;
+
+import java.sql.Connection;
+import java.sql.Driver;
+import java.sql.SQLException;
+import java.util.Properties;
+
+import org.apache.hive.jdbc.HiveDriver;
+import com.metabase.hive.jdbc.FixedHiveConnection;
+
+public class FixedHiveDriver extends HiveDriver {
+    static {
+        try {
+            java.sql.DriverManager.registerDriver(new FixedHiveDriver());
+        } catch (SQLException e) {
+            // TODO Auto-generated catch block
+            e.printStackTrace();
+        }
+    }
+
+    public FixedHiveDriver() {
+        super();
+    }
+
+    @Override
+    public Connection connect(String url, Properties info) throws SQLException {
+        return acceptsURL(url) ? new FixedHiveConnection(url, info) : null;
+    }
+}
```

Comment:
couldn't these classes just be written in Clojure? You can still extend classes and things like that. Take a look at https://clojure.org/reference/java_interop#_implementing_interfaces_and_extending_classes